### ISSUE TITLE
highway=path synonyms of highway=footway and cycleway should be rendered the same

### ISSUE THREAD

**Issue by kaneap** [organicmaps/organicmaps#1796]
As shown on this [OSM wiki page](https://wiki.openstreetmap.org/wiki/Path_examples), footways, cycleways, and birdleways can also be tagged as a highway path and additionally tagging foot=designated, bicycle=designated, or horse=designated respectively. Should these synonyms be displayed identically within organicmaps?

**Comment by biodranik** [comment]
Ideally, yes, differently tagged ways for the same purpose (either cycling, foot, or horse) should be rendered equally and properly used in routing. Do you have some good examples where we can quickly check it?

Also, the hardest question is what should be rendered when there are 2 or all 3 combinations of horse/cycling/foot present?

**Comment by pastk** [comment]
I'll add here `highway=footway` tagged with e.g. `sac_scale=mountain_hiking` - it should be displayed like `highway=path`.

> 
> Also, the hardest question is what should be rendered when there are 2 or all 3 combinations of horse/cycling/foot present?

FYI we do have a `[type.highway.path.bicycle]` "en = Cycle & Foot Path" in the types. Though I have no idea how it is rendered.

**Comment by kaneap** [comment]
In addition, would it make sense for an unpaved `highway=footway` to be rendered like `highway=path`?

I do think the proper rendering of a mixed use paths is the main issue here. Compare: https://www.openstreetmap.org/way/729942249 and https://www.openstreetmap.org/way/791858985. Also, I have noticed regional variation to this. Using `highway=path` for mixed use paths seems to be more popular in Germany than in the US, for example.

As for foot, compare https://www.openstreetmap.org/way/221577530 to any `highway=footway`

`highway=path` with `bicycle=designated` seems to probably mostly be mountain biking paths, so perhaps these should be rendered differently. However, without `surface` or `smoothness`, this is rather ambiguous.

As for what to do in the odd combination cases of horse, cycle, and foot, I think it makes most sense to focus on foot and cycle, simply due to the relevance to most people.

**Comment by biodranik** [comment]
> 'cause `surface=` is not imported into OM map files for paths (and roads too?) as far as I know.

surface tag is supported in our maps generator.

@kaneap can you please help and provide a detailed implementation proposal of what combinations of tags should be converted into what? And how they should be rendered? It may really help to speed up the proper fix.

**Comment by starsep** [comment]
> As shown on this [OSM wiki page](https://wiki.openstreetmap.org/wiki/Path_examples), footways, cycleways, and birdleways can also be tagged as a highway path and additionally tagging foot=designated, bicycle=designated, or horse=designated respectively. Should these synonyms be displayed identically within organicmaps?

I disagree with a statement that `highway=cycleway` and `highway=path bicycle=designated` are synonyms.
It seems kinda-implied by this quote, I'm sorry if that wasn't your intention :).
Let's not oversimplify, more tags needs to be considered as shown in OSM wiki page.
Let me share few examples of tagging in Poland.

|tagging|photo|
|---|---|
|`highway=cycleway`|![](https://upload.wikimedia.org/wikipedia/commons/4/4f/Droga_dla_rower%C3%B3w_Wis%C5%82ostrada.JPG)|
|`highway=path bicycle=designated foot=designated segregated=yes`|![](https://upload.wikimedia.org/wikipedia/commons/d/d3/Tunel_Augusta_Emila_Fieldorfa_w_Bia%C5%82ymstoku_1.jpg)|
`highway=path bicycle=designated foot=designated segregated=no`| ![](https://www.temi.pl/wp-content/uploads/2021/03/ciag-pieszo-rowerowy.jpg) | 

They are sorted from best to worst for cyclists.
Road signs are different (different laws/rules apply), they differ by bicycle/pedestrian separation.

They also have different `surface`, `smoothness`, and `width`.

**Comment by kaneap** [comment]
I agree, shared/segregated ways should be rendered differently. One issue, however, it that there is a bit of regional variation in this. A shared cycle/footpath is usually tagged as `highway=path bicycle=designated foot=designated segregated=no`, in Germany (and Poland, as you've shown), whereas `highway=cycleway bicycle=designated foot=designated segregated=no` is more popular in the US, for example.

**Comment by matkoniecz** [comment]
> Do you have some good examples where we can quickly check it?


`highway=path + bicycle=designated + foot=designated + segregated=yes + surface=paved` (can be treated as `highway=cycleway` https://www.openstreetmap.org/way/755634524#map=18/50.06894/19.94318 )

`highway=path + bicycle=designated + foot=designated + segregated=no + surface=ground` (unpaved, can be shown as path like it is shown now) https://www.openstreetmap.org/way/232175898#map=17/50.15419/19.86667

`highway=path + bicycle=designated + foot=designated + segregated=no + surface=paving_stones` ( can be treated as `highway=cycleway` https://www.openstreetmap.org/way/239890963#map=19/50.07638/19.91185

I can attach screenshoots - right now areas with paved segregated combined footway + cycleway look quite silly in Organic Maps.

**Comment by pastk** [comment]
> 
> I can attach screenshoots if that would be useful - right now areas with paved segregated combined footway + cycleway look quite silly in Organic Maps.

Yes, attach a screenshot please!

**Comment by matkoniecz** [comment]
added as links to prevent getting overly long thread measured in parsecs

**Comment by pastk** [comment]
We can map e.g. `highway=path bicycle=designated` into a feature with two OM types `highway-path` and `highway=cycleway`, this combination will be rendered with two overlapping lines (e.g. a dashed "trail" line and a solid violet cycleway line on top). Perhaps it'll look alright (need to test). E.g. see how a path together with a ski trail looks: https://github.com/organicmaps/organicmaps/issues/1577#issuecomment-1025788515

@vng are `highway-cycleway` and `highway-path hwtag-yesbicycle` different when it comes to routing?

Of course, as @starsep noted above, cycleway and path + bicycle=designated are not exactly the same. Still, rendering as two lines will be an improvement over the current situation (just the dashed trail line). A more detailed cycling paths classification could be done in a dedicated cycling style.

(to do this kind of mapping we need a "!=" matching operator support in `mapcss-mapping.csv`)

**Comment by vng** [comment]
Yes, hwtag-yesbicycle routing weight (speed) = 0.9 * highway-cycleway

A manually dedicated cycleway should be a little better than say footway with a cycleway=yes.

**Comment by pastk** [comment]
> Yes, hwtag-yesbicycle routing weight (speed) = 0.9 * highway-cycleway
> 
> A manually dedicated cycleway should be a little better than say footway with a cycleway=yes.

It will be good to distinguish `bicycle=yes` and `bicycle=designated` also, now both of them become `hwtag-yesbicycle` AFAIR

**Comment by matkoniecz** [comment]
> I disagree with a statement that highway=cycleway and highway=path bicycle=designated are synonyms

OK, what about more precise "`highway=path bicycle=designated segregated=yes`  has segment equivalent to `highway=cycleway`"?

Note that `highway=cycleway` also may have terrible surface (for example sett or uneven paving stones).

For example, this is `highway=cycleway`

![Parodia_drogi_dla_rowerów](https://github.com/organicmaps/organicmaps/assets/899988/9b0bbf5f-ac85-4887-9c57-08c3c5a52cfa)

**Comment by pastk** [comment]
> In addition, would it make sense for an unpaved `highway=footway` to be rendered like `highway=path`?

OSM's approach is to tag roads/paths according to their intended purpose, not to the physical appearance.
Hence a very rough hiking trail might be tagged as a `highway=footway` because its intended to be walked.
At the same time there could be a `highway-path` with `surface=asphalt smoothness=excellent` just because there is no sign (or other available info) that its designated for walking primarily.

Historically OM renders footways and paths very differently: `highway=footway` is a white dashed line while `highway=path` is a brown dashed line. And OM users usually expect white footways to be of good physical quality (paved, e.g. an urban sidewalk), while brown paths to be like unpaved trails.

IMO for most of the users it makes much more sense to render according to the physical appearance, not the intended use.

So its better to convert a `highway=path surface=paved` into a `highway-footway`, while a `highway=footway surface=unpaved` into a `highway=path`.
(if its a `highway=path surface=paved bicycle=yes/designated` then makes sense to convert into both `highway-footway` and `highway=cycleway`).

**Comment by VladimirMorozov** [comment]
It's also important to note that while many mappers prefer 
`highway=path bicycle=designated foot=designated`
for combined bicycle-pedestrian paths, the iD-editor creates
`highway=cycleway bicycle=designated foot=designated`

Both of these have the same meaning and should be rendered the same way.
And the question is in which way. We have a dotted white line and a solid pink line. One would hide such roads from pedestrians and the other one from cyclists.

**Comment by vng** [comment]
> (if its a highway=path surface=paved bicycle=yes/designated then makes sense to convert into both highway-footway and highway=cycleway)

And how we supposed to draw it? I think that the current highway-path-bicycle and hwtag-bicycle (for routing) are ok now.

And we can draw highway-cycleway and highway-path-bicycle the same.

**Comment by pastk** [comment]
> > (if its a highway=path surface=paved bicycle=yes/designated then makes sense to convert into both highway-footway and highway=cycleway)
> 
> And how we supposed to draw it? 

In two lines:
- a `highway-footway` below (a current wider white dashed line)
- a `highway=cycleway` above (a current thinner solid pink line)

I think it'll look alright and will be very recognizable.
Probably It'll look even better if those two lines a side-by-side, but we don't have this ability in the renderer yet and there might be confusion which side is for pedestrians and which is for cyclists (its often not tagged or there is just a single shared lane).

Another option is to develop a separate visual style for shared footways + cycleways (ideas?).

> I think that the current highway-path-bicycle + hwtag-paved + hwtag-bicycle (for routing) are ok now.

Its OK for routing indeed, but not for visually distinguishing paths / footways / cycleways (ATM it looks just as a brown dashed path).

**Comment by VladimirMorozov** [comment]
>Another option is to develop a separate visual style for shared footways + cycleways (ideas?).

In additon to drawing cyclway on top of dashed line, maybe check out how these two will look like:
1) pink dashed line, same width as a footway
2) white dashed line with pink non-transparent gaps, same width as a footway.

**Comment by pastk** [comment]
> IMO for most of the users it makes much more sense to render according to the physical appearance, not the intended use.
> 
> So its better to convert a `highway=path surface=paved` into a `highway-footway`, while a `highway=footway surface=unpaved` into a `highway=path`.

I think we can convert following `highway=footway` into `highway=path`:
- unpaved/unformed surface
- `smoothness=bad` or worse
- `tracktype=grade3-5`
- `sac_scale=*` (implemented already)
- `informal=yes`
It could convert up to 3%.
When converting we need to transfer implied access mode accordingly, e.g. by default set `bicycle=no`.

Converting `highway=path` to `highway=footway`:
- paved surface (probably including `surface=compacted` which is not really "paved" but is formed and good quality)
- `smoothness=intermediate` or better
- `tracktype=grade1/2`
- `footway=*` (sidewalk, crossing...)
- `lit=yes`
It could convert up to 10% of all paths into footways.
Probably set `bicycle=yes` by default..

**Comment by matkoniecz** [comment]
I would skip converting `informal=yes` `smoothness` ones.

Do we really want `surface=paved informal=yes` to be shown like unpaved ones? Or `surface=dirt smoothness=intermediate` as paved?

Also, dirt path with `surface=sidewalk` or `=crossing` should not be upgraded to footway.

>  When converting we need to transfer implied access mode accordingly, e.g. by default set bicycle=no.

Are you sure? I would not assume `bicycle=no` on all footways (though I would give noticeable penalty in routing)

**Comment by pastk** [comment]
Some attribs (e.g. surface) should have a priority over other, of course. The exact algo is not defined yet.
BTW had this work been done for the osm.org standard layer already?

Re access modes need to check how the OM router treats the difference between footways and paths at the moment. The idea is that "conversion" shouldn't affect routing.

**Comment by matkoniecz** [comment]
>  BTW had this work been done for the osm.org standard layer already?

I did it. It considers solely `surface` (paved vs unpaved) and nothing else.

**Comment by pastk** [comment]
> Do we really want `surface=paved informal=yes` to be shown like unpaved ones? Or `surface=dirt smoothness=intermediate` as paved?
> 
> Also, dirt path with `surface=sidewalk` or `=crossing` should not be upgraded to footway.

- please take a look at the suggested algo in https://github.com/organicmaps/organicmaps/pull/7878

**Comment by pastk** [comment]
re **bridleways**

There are only ~100k of them and usually they can be used by walkers and cyclists too.

Whereas e.g. `horse=yes/designated` is 600k and `horse=no` is 1M.

So maybe suitability for horses should be displayed as an overlay/underlay (ideally optional) of more common path types (white footway, brown path, cycleway line)..

**Comment by pastk** [comment]
> On yellow (fields) and green (forests) background it becomes gray for my eyes. 

Yeap it doesn't look good, thanks for reporting!

> 
> I suggest to try making cycleway beneath footpaths more opaque and same width as white lines.

I've tried making it same width, but there were complaints that cycleways are too promiment.
Opaqueness was reduced a lot for the same reason...
Maybe it could be increased but on high/detailed zooms only.

Details:
- https://github.com/organicmaps/organicmaps/pull/7925
- https://github.com/organicmaps/organicmaps/pull/8045
- https://github.com/organicmaps/organicmaps/issues/8529
